packages/security/src/CredentialManagerOverride.ts

```diff
@@ -0,0 +1,231 @@
+/*
+* This program and the accompanying materials are made available under the terms of the
+* Eclipse Public License v2.0 which accompanies this distribution, and is available at
+* https://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Copyright Contributors to the Zowe Project.
+*
+*/
+
+import * as path from "path";
+import { readJsonSync, writeJsonSync } from "fs-extra";
+
+import { ICredentialManagerNameMap } from "./doc/ICredentialManagerNameMap";
+import { ImperativeConfig } from "../../utilities";
+import { ImperativeError } from "../../error";
+import { ISettingsFile } from "../../settings/src/doc/ISettingsFile";
+
+/**
+ * This class provides access to the known set of credential manager overrides
+ * and functions to manipulate which credential manager is in use.
+ * Other credential managers can replace the default credential manager.
+ * Both CLI plugins and Zowe Explorer extensions can override the default
+ * credential manager. However, only one credential manager will be in effect
+ * on a given computer. The last component to override the credential
+ * manager wins.
+ */
+export class CredentialManagerOverride {
+    public static readonly CRED_MGR_SETTING_NAME: string = "CredentialManager";
+    public static readonly DEFAULT_CRED_MGR_NAME: string = "@zowe/cli";
+
+    private static readonly KNOWN_CRED_MGRS: ICredentialManagerNameMap[] = [
+        {
+            "credMgrDisplayName": this.DEFAULT_CRED_MGR_NAME
```

Comment:
Should the default display name be `"Zowe CLI"` if we want it to match the style of other display names? I'm fine with leaving it as `"@zowe/cli"` for backward compatibility 🙂